### ISSUE TITLE
calling function returned by a function

### ISSUE THREAD

**Issue by DevrosTheOne** [vlang/v#10117]
**V version:** 
V full version: V 0.2.2 e657cb2.d5c306c

**OS:** 
OS: macos, macOS, 11.3.1, 20E241
Processor: 8 cpus, 64bit, little endian, Apple M1
CC version: Apple clang version 12.0.5 (clang-1205.0.22.9)

**What did you do?**
fn gen_fn() fn (int, int) int {
	return fn (a int, b int) int {
		return a + b
     }
}

println(gen_fn()(5, 4))
// f := gen_fn()
// println( f(5, 4) )

**What did you expect to see?**
9
 
**What did you see instead?**
fungen.v:7:16: error: unexpected token `,`, expecting `)`
    5 | }
    6 | 
    7 | x := gen_fn()(5, 4)
      |                ^

**Comment by dumblob** [comment]
I'm afraid what you're actually asking for are [closures](https://github.com/vlang/v/issues/7740 ) which do not seem to be implemented yet.

**Comment by Gladear** [comment]
No, closures aren't involved here. This is a parser issue, this code should be valid (even if it is quite unreadable imho...)

**Comment by DevrosTheOne** [comment]
Not a closure, nothing is scoped in the function. 
V has first class functions so I was assuming this would be a feature.
Language very minimal in the design so things can be a bit tight.  Might have gone a bit spartan in the example :sunglasses:

**Comment by dumblob** [comment]
@Gladear @DevrosTheOne both the parsing of "functions in functions" constructs as well as the actual analysis whether there is something scoped seem to fall under the "closures" umbrella. But maybe I'm (again :wink:) out of sync with latest *V* development.

On the other hand I skipped the error *V* currently spits out. And now I see what @Gladear meant - because the complaint is about the syntax `f()()` which I thought is already supported.
